src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -1654,12 +1682,53 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
         {
             base.AfterAddingTypeMembersChecks(conversions, diagnostics);
 
+            var compilation = this.DeclaringCompilation;
+            var location = locations[0];
+
             if (IsDeclaredReadOnly && !ContainingType.IsReadOnly)
             {
-                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, locations[0], modifyCompilation: true);
+                compilation.EnsureIsReadOnlyAttributeExists(diagnostics, location, modifyCompilation: true);
+            }
+
+            if (compilation.ShouldEmitNullableAttributes(this) &&
+                ShouldEmitNullableContextValue(out _))
+            {
+                compilation.EnsureNullableContextAttributeExists(diagnostics, location, modifyCompilation: true);
             }
         }
 
+        internal override byte? GetLocalNullableContextValue()
```

Comment:
Any reason not to push this down to SourceMethodSymbol?

Reply:
We could move this to `SourceMethodSymbol` and that would enable emitting `NullableContextAttribute` on local functions and lambdas, but that would probably require storing this state in a new field. Logged [https://github.com/dotnet/roslyn/issues/36736](https://github.com/dotnet/roslyn/issues/36736) and added a comment here.

---
In reply to: [296959203](https://github.com/dotnet/roslyn/pull/36152#discussion_r296959203) [](ancestors = 296959203)